ci/github_release.py

```diff
@@ -0,0 +1,223 @@
+import os
+
+from string import Template
+
+import ctx
+import glci.model
+import product.v2
+import release
+
+
+def _find_path_in_manifest(
+    suffix: str,
+    manifest: glci.model.OnlineReleaseManifest,
+) -> glci.model.S3_ReleaseFile:
+    return next(path for path in manifest.paths if path.suffix == suffix)
+
+
+def _find_manifest_in_set(
+    platform: str,
+    manifest_set: glci.model.OnlineReleaseManifestSet,
+) -> glci.model.OnlineReleaseManifest:
+    return next(manifest for manifest in manifest_set.manifests if manifest.platform == platform)
+
+
+def _get_download_url(
+    suffix: str,
+    platform: str,
+    image_metadata_tag: str,
+    manifest_set: glci.model.OnlineReleaseManifestSet,
+):
+    manifest = _find_manifest_in_set(
+        platform=platform,
+        manifest_set=manifest_set,
+    )
+    if manifest == None:
+        print('Could not find release-manifest for platform {manifest}, skipping release.')
+        return None
+
+    path = _find_path_in_manifest(suffix, manifest)
+    if path == None:
+        print(f'Could not find path for platform {platform=}, skipping release.')
+        return None
+
+    return (
+        path.name,
+        f'https://gardenlinux.s3.eu-central-1.amazonaws.com/{path.s3_key}',
+        getattr(manifest.published_image_metadata, image_metadata_tag),
+    )
+
+
+def get_manifest(
+    cicd_cfg_name: str,
+    committish: str,
+    flavourset_name: str,
+    gardenlinux_epoch: str,
+    version: str,
+) -> bool: # True on success else False:
+
+    cicd_cfg = glci.util.cicd_cfg(cfg_name=cicd_cfg_name)
+    if not (package_build_cfg := cicd_cfg.package_build):
+        raise RuntimeError(f"No package-build config found in cicd-config {cicd_cfg_name}")
+
+    aws_cfg_name = cicd_cfg.build.aws_cfg_name
+    s3_bucket_name = package_build_cfg.s3_bucket_name
+
+    print(f'downloading release manifest from s3 {aws_cfg_name=} {s3_bucket_name=}')
+
+    find_release_set = glci.util.preconfigured(
+        func=glci.util.find_release_set,
+        cicd_cfg=glci.util.cicd_cfg(cicd_cfg_name),
+    )
+
+    manifest = find_release_set(
+        flavourset_name=flavourset_name,
+        build_committish=committish,
+        version=version,
+        gardenlinux_epoch=gardenlinux_epoch,
+        build_type=glci.model.BuildType.RELEASE,
+        absent_ok=True,
+    )
+    return manifest
+
+
+def make_release(
+    cicd_cfg_name: str,
+    ctx_repository_config_name: str,
+    committish: str,
+    flavourset_name: str,
+    gardenlinux_epoch: str,
+    giturl: str,
+    version: str,
+    repo_dir: str,
+):
+    manifest_set = get_manifest(
+        cicd_cfg_name=cicd_cfg_name,
+        committish=committish,
+        flavourset_name=flavourset_name,
+        gardenlinux_epoch=gardenlinux_epoch,
+        version=version,
+    )
+    if not manifest_set:
+        print('Could not find release-manifest set, skipping release.')
+        return False
```

Comment:
after a cursory check: I think this function returns either `False` or `None`, nothing else. Consider returning `True` in case of success.